expressions_api/expression_verify.go

```diff
@@ -0,0 +1,143 @@
+type ExpressionVerificationInput struct {
+	spannerClient    spannerclient.SpannerClient
+	expressionDetail internal.ExpressionDetail
+}
+
+func (ev *ExpressionVerificationAccessorImpl) VerifyExpressions(ctx context.Context, verifyExpressionsInput internal.VerifyExpressionsInput) internal.VerifyExpressionsOutput {
+	err := ev.validateRequest(verifyExpressionsInput)
+	if err != nil {
+		return internal.VerifyExpressionsOutput{Err: err}
+	}
+	dbURI := ev.SpannerAccessor.SpannerClient.DatabaseName()
+	dbExists, err := ev.SpannerAccessor.CheckExistingDb(ctx, dbURI)
+	if err != nil {
+		return internal.VerifyExpressionsOutput{Err: err}
+	}
+	if dbExists {
+		err := ev.SpannerAccessor.DropDatabase(ctx, dbURI)
+		if err != nil {
+			return internal.VerifyExpressionsOutput{Err: err}
+		}
+	}
+	verifyExpressionsInput.Conv, err = ev.removeExpressions(verifyExpressionsInput.Conv)
+	if err != nil {
+		return internal.VerifyExpressionsOutput{Err: err}
+	}
+	err = ev.SpannerAccessor.CreateDatabase(ctx, dbURI, verifyExpressionsInput.Conv, verifyExpressionsInput.Source, constants.DATAFLOW_MIGRATION)
+	if err != nil {
+		return internal.VerifyExpressionsOutput{Err: err}
+	}
+	//Drop the staging database after verifications are completed.
+	defer ev.SpannerAccessor.DropDatabase(ctx, dbURI)
+	verificationInputList := make([]ExpressionVerificationInput, len(verifyExpressionsInput.ExpressionDetailList))
+	for i, expressionDetail := range verifyExpressionsInput.ExpressionDetailList {
+		verificationInputList[i] = ExpressionVerificationInput{
+			spannerClient:    ev.SpannerAccessor.SpannerClient,
+			expressionDetail: expressionDetail,
+		}
+	}
+	r := task.RunParallelTasksImpl[ExpressionVerificationInput, internal.ExpressionVerificationOutput]{}
+	expressionVerificationOutputList, _ := r.RunParallelTasks(verificationInputList, THREAD_POOL, ev.verifyExpressionInternal, true)
+	var verifyExpressionsOutput internal.VerifyExpressionsOutput
+	var errorCount int16 = 0
+	for _, expressionVerificationOutput := range expressionVerificationOutputList {
+		verifyExpressionsOutput.ExpressionVerificationOutputList = append(verifyExpressionsOutput.ExpressionVerificationOutputList, expressionVerificationOutput.Result)
+		if expressionVerificationOutput.Result.Err != nil {
+			errorCount++
+		}
+	}
+	if errorCount != 0 {
+		verifyExpressionsOutput.Err = fmt.Errorf("%d expressions either failed verification or did not get verified. Please look at the individual errors returned for each expression", errorCount)
+
+	}
+	return verifyExpressionsOutput
+}
+
+func (ev *ExpressionVerificationAccessorImpl) verifyExpressionInternal(expressionVerificationInput ExpressionVerificationInput, mutex *sync.Mutex) task.TaskResult[internal.ExpressionVerificationOutput] {
+	var sqlStatement string
+	switch expressionVerificationInput.expressionDetail.Type {
+	case "CHECK":
```

Comment:
I would suggest having these added to the constant file and referring it back from there to avoid errors

Reply:
Done